server/workflow/workflow_server.go

```diff
@@ -74,12 +74,16 @@ func (s *workflowServer) CreateWorkflow(ctx context.Context, req *workflowpkg.Wo
 	}
 
 	wf, err := wfClient.ArgoprojV1alpha1().Workflows(req.Namespace).Create(req.Workflow)
-
 	if err != nil {
-		log.Errorf("Create request is failed. Error: %s", err)
+		if apierr.IsServerTimeout(err) && req.Workflow.GenerateName != "" && req.Workflow.Name != "" {
+			errWithHint := fmt.Errorf(`create request failed due to timeout, but it's possible that workflow "%s" already exists. Original error: %w`, req.Workflow.Name, err)
+			log.Error(errWithHint)
```

Comment:
why log here? why at error? not an error?

Reply:
This kind of log is present elsewhere in the server, so I added it to keep it consistent. We log at error because it is: the request timed out, we just happen wrap the error with a hint.